### PR TITLE
Global: Implement Dark Mode 🌙

### PR DESCRIPTION
**Pull request by Samk13**
:heart: Thank you for your contribution!

### Description

This pull request includes the following changes:

* Set up dark mode 🌙
* Remove outdated overrides
* Auto pick prefers-color-scheme from system settings
* Resolves the asset URL problem failing CI by removing the base override.

These changes enhance the user experience by providing an alternative color scheme that is easier on the eyes in low-light environments.

Screenshots:

image

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Comment by fenekku** [comment]
Thanks for that.

When I run it here is what I get on the frontpage by default:

frontpage_default_light

... where the dark mode toggle is missing. Let's have it there too. It's also missing on the Features page.

Then when I click on the "Install" tab, it switches to dark mode automatically:

install_default_dark

And then when I click on the home logo to go back to the frontpage:

frontpage_default_dark

...it switches to dark mode.

I suspect the frontpage starts with a value unset. The pages that have the dark mode toggle set that value, and so when visiting the frontpage after, the value is now set and the dark theme kicks in. My slightly quirky setup may be revealing that issue: Ubuntu 22.04 light mode + Firefox automatic mode + Mozilla dark theme addon enabled. (For comparison this didn't happen on Brave (Chromium based) browser on the same platform). 

Getting dark mode setting for front+feature page should provide a consistent experience.

File: docs/stylesheets/extra.css [hunk 1]
```diff
@@ -1,208 +1,258 @@
...
-    color: #000000;
-    opacity: 0.54;
-}
+/* Color variables
+https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/assets/stylesheets/main/_colors.scss
+*/
+:root {
+    --color-invenio: #0377cd;
+    --color-black: #000000;
+    --color-white: #ffffff;
```
[the remark this answers is not in the document]
Comment (Samk13) [reply docs/stylesheets/extra.css:7]: I've thought about it, but using `--main-light` and `--main-dark` suggests a relation to the main theme, which isn't the case here. The purpose, as you mentioned, is to have a single point of control for easy color management. Custom changes can still be made directly when needed.

Another point is that much of the `extra.css`  is redundant and can be minimized even more. I'd prefer to keep it as is for now, and in a future PR, we could remove more of the custom rules altogether.

**Comment by Samk13** [comment]
> I suspect the frontpage starts with a value unset. The pages that have the dark mode toggle set that value, and so when visiting the frontpage after, the value is now set and the dark theme kicks in. My slightly quirky setup may be revealing that issue: Ubuntu 22.04 light mode + Firefox automatic mode + Mozilla dark theme addon enabled. (For comparison this didn't happen on Brave (Chromium based) browser on the same platform).
> 
> Getting dark mode setting for front+feature page should provide a consistent experience.

As we discussed in the DMs, the reason the toggle is not showing is that the [current inheritance](https://github.com/inveniosoftware/docs-invenio-rdm/blob/master/theme/frontpage_base.html) doesn't support displaying the toggle and search input on the front page without routing in the docs. You can simply replace one line `{% extends "main.html" %}` [here](https://github.com/inveniosoftware/docs-invenio-rdm/blob/master/theme/frontpage.html#L1), but then we need to refactor these pages and rewrite the CSS again. 
However, as you mentioned, we could address this in another PR unless it is urgently needed.

**Comment by Samk13** [comment]
Here are my findings  regarding the override pages (Home, features):

- [Overriding base.html](https://squidfunk.github.io/mkdocs-material/customization/?h=exte#extending-the-theme) causes the theme toggle and search field to disappear from the header. Removing the override restores them, which is confusing.
- The [hero block](https://github.com/squidfunk/mkdocs-material/blob/master/material/templates/base.html#L144) needs to be moved under the navigation tab in the base template for the wave effect to display correctly but the previous issue is preventing us doing so.

We can discuss further in DM @fenekku

**Comment by Samk13** [comment]
Update:
- Fix the toggle not showing on front-page and features.
- Adapt the style without overriding `base.html`

Screenshot from 2024-06-22 04-28-09
Screenshot from 2024-06-22 04-28-00

**Review by fenekku — APPROVED** [review]
Awesome! So glad we got 

- dark mode
- search on frontpage
- break from replacing base.html completely and therefore URL of assets problem fixed

Thanks for all the effort on this one!

**Comment by fenekku** [comment]
Final look:

image

**Pull request merged**